### DISCUSSION TITLE
If we were to provide an analysis of the source code for a mobile app?

### DISCUSSION THREAD

**Discussion by hahwul** [owasp-noir/noir#159] [Ideas]
Noir currently supports source code analysis for web-based applications. 
There has been demand for analysis of mobile apps for some time, and now it's time to organize our thoughts on this.

If we were to offer analysis for mobile app's source code, it seems we would focus more on the App's URI Scheme/DeepLink rather than the Web App Endpoint. URI Scheme and Deep Links are crucial, but would this align with Noir's way?

**Comment by hahwul** [comment]
Hey @ksg97031, let's revisit this issue. The method might be challenging, but I believe it's a crucial function. 🤔

**Reply by ksg97031** [reply]
@hahwul Could this approach be implemented using an LLM by first extracting the Exported Components, analyzing the Intent source code, and then iteratively requesting the relevant parameters and schemes? 🤔

This could be one of the better approaches for an LLM to handle complex source code.

**Reply by hahwul** [reply]
@ksg97031 
Of course, using an LLM seems like the easiest solution. However, I'm currently developing the LLM integration(#473) , and on my personal MacBook, it's not incredibly fast, haha. I'll think about the approach a bit more!
